File: python/consumer.py
from kafka import KafkaConsumer
import json

print("Consumer...")

# Define the Kafka topics to subscribe to
topics = ["test", "hello-world", "goodbye-world"]

# Configure the Kafka consumer
consumer = KafkaConsumer(
    *topics,
    bootstrap_servers=["localhost:9092"],
    auto_offset_reset="earliest",
    enable_auto_commit=True,
    group_id="test-group",
    value_deserializer=lambda x: json.loads(x.decode("utf-8"))
)

# Main loop - listen for messages to subscribed topics
try:
    for message in consumer:
        print(f"Received message: {message.value}")
except KeyboardInterrupt:
    print("Consumer stopped by user")
except Exception as e:
    print(f"An error occurred: {str(e)}")
finally:
    consumer.close()

File: nodejs/consumer.js

console.log("Sonsumer...");

import { Kafka } from 'kafkajs';

const kafka = new Kafka({
    clientId: "kafka",
    brokers: ["localhost:9092"],
});

// Define the topics to subscribe to
const topics = ["test", "hello-world", "goodbye-world"]

const consumer = kafka.consumer({groupId: "test-group"});

async function run() {
    await consumer.connect();

    // Subscribe to multiple topics
    await consumer.subscribe({
        topics: topics,
        fromBeginning: true
    });

    await consumer.run({
        eachMessage: async ({ topic, partition, message }) => {
            console.log(`Received message: ${message.value} from topic: ${topic}`)
        },
    });
}

run().catch(console.error);
